File: UVA OJ/11292.cc
//Result:Wizmann	UVA 11292	Accepted	0 KB	48 ms	C++ 4.5.3	854 B	2013-01-15 20:52:09
#include <cstdio>
#include <cstdlib>
#include <cstring>
#include <iostream>
#include <algorithm>
#include <queue>

using namespace std;

#define print(x) cout<<x<<endl
#define input(x) cin>>x
#define SIZE 20480
#define pq priority_queue

int main()
{
	freopen("input.txt","r",stdin);
	int n,m,t;
	pq< int,vector<int>,greater<int> > head,knight;

	while(input(n>>m) && n+m)
	{
		while(!head.empty()) head.pop();
		while(!knight.empty()) knight.pop();
		for(int i=0;i<n;i++)
		{
			input(t);
			head.push(t);
		}
		for(int i=0;i<m;i++)
		{
			input(t);
			knight.push(t);
		}

		int ok=0,cost=0;
		while(!head.empty() && !knight.empty())
		{
			int hh=head.top();
			head.pop();
			int kk=knight.top();
			knight.pop();

			if(hh>kk)
			{
				head.push(hh);
			}
			else
			{
				ok++;
				cost+=kk;
			}
		}

		if(ok==n) print(cost);
		else print("Loowater is doomed!");
	}
	return 0;
}

File: UVA OJ/11292.py
#python3

'''LONG LIVE PYTHON2'''
xrange = range
__map = map
map = lambda func, ns: list(__map(func, ns))
'''LONG LIVE PYTHON2'''

while True:
    n, m = map(int, input().split())
    if n + m == 0:
        break

    ns = map(int, [input() for i in xrange(n)])
    ms = map(int, [input() for i in xrange(m)])

    ns.sort()
    ms.sort()

    pn, pm = 0, 0
    kill = 0
    cost = 0
    while pn < n and pm < m:
        if ms[pm] >= ns[pn]:
            kill += 1
            cost += ms[pm]
            pm += 1
            pn += 1
        else:
            pm += 1
    if kill == n:
        print(cost)
    else:
        print('Loowater is doomed!')

'''
^^^^^TEST^^^^^
2 3
5
4
7
8
4
2 1
5
5
10
0 0
--------
11
Loowater is doomed!
$$$TEST$$$$$
'''



